### ISSUE TITLE
Transition regular store activity notices to snackbar component

### ISSUE THREAD

**Issue by elizaan36** [woocommerce/woocommerce#29239]
**Is your feature request related to a problem? Please describe.**
For more cohesiveness across WooCommerce and WordPress, and to clean up the UI for regular store management activity feedback, we should transition the legacy core WP notice style to the new snackbar component ([in Gutenberg Storybook here](https://wordpress.github.io/gutenberg/?path=/story/components-snackbar--default).)

**Existing notices:**
<img width="1068" alt="Screen Shot 2021-02-26 at 12 48 59 PM" src="https://user-images.githubusercontent.com/5121465/109343169-83363d80-7832-11eb-91c0-cef695ef67d9.png">
<img width="1065" alt="Screen Shot 2021-02-26 at 12 54 21 PM" src="https://user-images.githubusercontent.com/5121465/109343176-85000100-7832-11eb-90d9-d0e971a9d537.png">
<img width="1071" alt="Screen Shot 2021-02-26 at 12 56 21 PM" src="https://user-images.githubusercontent.com/5121465/109343182-87faf180-7832-11eb-970b-0d7f4816d096.png">

**Describe the solution you'd like**
Using the Gutenberg snackbar component, the `Order updated` (which is also used for when an order is created), `Product published`, and `Product updated` messages would appear as a snackbar that disappears after 5 seconds. For the product notices, the snackbar contains a link to view the product on the front end.

![Frame 2061](https://user-images.githubusercontent.com/5121465/109344844-e88b2e00-7834-11eb-8477-d28cf835a588.png)

Also, as a nice-to-have we should change "Product published" to "Product created" as that's the terminology used for creating products.

Example of the `Product updated` notice in the snackbar component rather than the banner notice style.
![Frame 58](https://user-images.githubusercontent.com/5121465/109343981-bb8a4b80-7833-11eb-888e-070dd4b52801.png)

Are there any notices I've missed for regular store activity that could transition to the snackbar component as well? The banner notice style will soon be deprecated so will be good to get ahead of the curve on that.

cc @pmcpinto

**Comment by juliaamosova** [comment]
Hi @elizaan36,

Thank you for highlighting this request and providing details of what is expected. 

We might need to do an overview of all the core features to see where we have inconsistencies and correct them. One of them that comes to mind is updating the old purple color to new purple on the CSV Products Import page:

![old_purple](https://user-images.githubusercontent.com/19143190/109347406-22623180-7841-11eb-9680-ba240d9b9f49.png)

I am cc'ying other WooCommerce Core leads here to put it on their radar @peterfabian @ObliviousHarmony @vedanshujain.

**Comment by elizaan36** [comment]
> We might need to do an overview of all the core features to see where we have inconsistencies and correct them. One of them that comes to mind is updating the old purple color to new purple on the CSV Products Import page

That's a great call out and it's been on my mind as well. Do you think for this specific issue it would be best to focus on the notices and I can follow up with another issue to update that color? In the interim before we start looking at a brand new product management experience, the progress bar and primary action shown in the screenshot would ideally pull in the WordPress theme color that user has chosen.

There's another issue on my mind that I might bring over to this repo - the product management guide tooltips are hiding some areas of the navigation on mobile: https://github.com/woocommerce/woocommerce-admin/issues/6250. I can create a new issue for that as well.

**Comment by juliaamosova** [comment]
>Do you think for this specific issue it would be best to focus on the notices and I can follow up with another issue to update that color?

Yep, I think creating a separate issue for updating the color will be best so that we could focus on the notices here. Thanks @elizaan36! 

>There's another issue on my mind that I might bring over to this repo - the product management guide tooltips are hiding some areas of the navigation on mobile: woocommerce/woocommerce-admin#6250. I can create a new issue for that as well.

Creating a separate issue for this will be great as ell, thank you @elizaan36!

**Comment by pmcpinto** [comment]
Routing this to product for the next step: the report needs a product decision before it becomes implementation work.
